src/handler-context/Config/index.ts

```diff
@@ -0,0 +1,69 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Json } from "../../shared/types";
+import { ConfigClient, ConfigFileNames, ConfigFiles } from "./types";
+
+export class Config<TFileName extends ConfigFileNames> {
+  private readonly _client: ConfigClient;
+  private readonly _files: ConfigFileNames[];
+  private readonly _logger: Logger;
+  private _hasCacheBeenPopulated: boolean = false;
+  private _cache!: Omit<ConfigFiles, keyof Omit<ConfigFiles, TFileName>>;
+  private _promises!: Array<
+    Promise<{
+      file: Json;
+      fileName: ConfigFileNames;
+    }>
+  >;
+
+  constructor(client: ConfigClient, files: TFileName[], logger: Logger) {
+    this._client = client;
+    this._files = files;
+    this._logger = logger;
+    this._populatePromises();
+  }
+
+  private readonly _populatePromises = (): void => {
+    if (process.env.CONFIG_BUCKET === undefined)
+      throw new Error("No CONFIG_BUCKET defined in this environment");
+
+    this._promises = this._files.map(async (fileName) => {
+      const promise = (async (): Promise<{
+        file: Json;
+        fileName: ConfigFileNames;
+      }> => {
+        const file = await this._client.getConfigFile(fileName);
+        return { file, fileName };
+      })();
+      return await promise;
+    });
+  };
+
+  public readonly populateCache = async (): Promise<void> => {
+    const resolutions = await Promise.allSettled(this._promises);
+    // @ts-expect-error
+    this._cache = resolutions.reduce<
+      Omit<ConfigFiles, keyof Omit<ConfigFiles, TFileName>>
```

Comment:
This seems like a very complex type. Why is that necessary?

Reply:
It's kind of like [`Pick`](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) but it works with an enum, unfortunately `Pick<ConfigFileNames, keyof TFileName>` is not valid. This type is necessary because it scopes in the type of the cached config to just the requested files, rather than all the files that could possibly be there. So if you build a handler context asking for just the vendor services and the rates configs then the resulting config cache type will be just

```
{
  services: Json;
  rates: Json;
}
```

Reply:
I did need this in 3 places though, I'll declare it and reuse it